src/map/entities/battleentity.cpp

```diff
@@ -282,17 +302,20 @@ uint8 CBattleEntity::GetSpeed()
     uint8 mazurkaQuickeningEffect = std::clamp<uint8>(getMod(Mod::MOVE_SPEED_QUICKENING) + getMod(Mod::MOVE_SPEED_MAZURKA), 0, 10);
 
     // We have all the modifiers needed. Calculate final speed.
-    // This MUST BE DONE IN THIS ORDER. Using uint8 data type, we use that to floor.
+    // This MUST BE DONE IN THIS ORDER. Using int8 data type, we use that to floor.
     outputSpeed = baseSpeed + additiveMods;
```

Comment:
This will need a clamp. This should not be going bellow 0, ever.

Reply:
It does. Retail testing:
```
+14 from bolters roll turns into +10 at -54 movement speed
-54 gear + 15 bolters + flee = 7 speed (-4 * 2 + 15)
```

Reply:
Gravity is also effected by this. With negative speed + bolter's roll, gravity will actually increase your speed.

Reply:
And that is the exact point I will just ignore retailaccuracy in favor of sanity

Reply:
The most you can get from gear is -65 speed, so the max negative you can get would be like -39 with cheer, gear and flee. Which will ultimately be 0. I don't think there's any danger of abuse. We're talking about a 1-2 speed difference from gravity in this case.

Reply:
Let's say we have -65 from gear, -99% gravity, +3% cheer, +24% gear, +100% flee, +36 max bolter's roll (+10 from XI roll, +10 crooked cards, +16 Rostam), +10 quickening. You're looking at a final 54 speed. But consider the setup for that. Taking any attack will remove bolter's and quickening, bringing you back down to 0. You would have to get a stronger gravity than anything actually does and then roll a perfect roll all while wearing no defensive gear to see any effect, and at most you're still running at basically normal speed. I'm more interested in an accurate emulator than preventing some infinitesimally small chance of abuse.

Reply:
Not exactly my point, just poorly worded on my end.

My point is, I dont believe SE actually intended for gravity to ever increase your speed. I believe its and edge case that results in ununtended behavior on their end.

Reply:
It really only happens because bolter's roll is unconditionally applied. We _could_  add a condition similar to quickening where it's only applied when speed >= 0, but I still feel like it's unnecessary. I'm a big believer in emulation accuracy, faults and all.